src/parser.rs

```diff
@@ -241,7 +243,7 @@ fn parse_timestamp(m: &str) -> ParseResult<(Option<time_t>, &str)> {
     tm.tm_sec = take_item!(parse_num(rest, 2, 2), rest);
     if rest.starts_with('.') {
         take_char!(rest, '.');
-        take_item!(parse_num(rest, 1, 6), rest);
+        tm.tm_nsec = take_item!(parse_num(rest, 1, 6), rest) * NSEC_PER_MS;
```

Comment:
technically this can have up to microsecond precision, right? so you need to look at how many digits you actually got back...